internal/lock/lock.go

```diff
@@ -0,0 +1,189 @@
+package lock
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/url"
+
+	"go.etcd.io/etcd/clientv3"
+)
+
+const (
+	keyTemplate  = "com.coreos.airlock/groups/%s/v1/semaphore"
```

Comment:
@arithx should I make the key-prefix here (`com.coreos.airlock`) configurable? If so, should it be at runtime (via a TOML entry) or at build time (a var that can be overridden via link-flags)?

Reply:
@lucab do you have any scenarios in mind where the user might want to configure the key-prefix? I tried coming up with a few but the only one I had was if the user had copies of the same cluster in different environments and wanted to keep the same keynames, but at that point there should also be a separate etcd server. cc @bgilbert 

If we do make it configurable I'd lean towards run-time configurable over build-time.

Reply:
No specific scenario (maybe unbranding, but feels quite stretched). I'm fine with keeping this hardcoded.